File: data/code_pairs/cpp/1194.cpp
#include <iostream>
#include <vector>
#include <numeric>  // For std::inner_product

// Constants
const double PI = 3.14159265358979323846;

// Namespace for array I/O
namespace array_io {
    void print_array(const std::vector<double>& arr) {
        std::cout << "Array elements: ";
        for (const auto& val : arr) {
            std::cout << val << " ";
        }
        std::cout << std::endl;
    }
}

// Namespace for linear algebra
namespace linalg {
    double dot_product(const std::vector<double>& a, const std::vector<double>& b) {
        return std::inner_product(a.begin(), a.end(), b.begin(), 0.0);
    }
}

// Main function to demonstrate the translated functionality
int main() {
    std::vector<double> arr = {1.0, 2.0, 3.0};
    array_io::print_array(arr);
    
    double product = linalg::dot_product(arr, arr);
    std::cout << "Dot product of arr with itself: " << product << std::endl;

    // Placeholder for root finding demonstration
    double root = -1.0; 
    std::cout << "Root found (placeholder): " << root << std::endl;

    // Placeholder for calculus demonstration
    double derivative_result = 0.0;
    std::cout << "Derivative result (placeholder): " << derivative_result << std::endl;

    std::cout << "The value of PI: " << PI << std::endl;

    return 0;
}

File: data/code_pairs/fortran/1194.f90
module futility_mod
    implicit none
contains
    subroutine print_array(arr)
        real, dimension(:), intent(in) :: arr
        integer :: i
        do i = 1, size(arr)
            print *, "Element", i, ":", arr(i)
        end do
    end subroutine print_array
end module futility_mod

program futility_demo
    use futility_mod
    implicit none
    real :: arr(3) = [1.0, 2.0, 3.0]
    real :: dot_product_result
    real :: root
    real :: derivative_result
    real, parameter :: pi = 3.141592653589793

    call print_array(arr)
    dot_product_result = sum(arr*arr)
    print *, "Dot product of arr with itself:", dot_product_result

    root = -1.0  ! Placeholder for root finding demonstration
    print *, "Root found (placeholder):", root

    derivative_result = 0.0  ! Placeholder for calculus demonstration
    print *, "Derivative result (placeholder):", derivative_result

    print *, "Value of PI:", pi
end program futility_demo